DependencyInjection/LiipMonitorExtension.php

```diff
@@ -240,18 +239,11 @@ private function configureDoctrineMigrationsCheck(ContainerBuilder $container, a
     private function configureMailer(ContainerBuilder $container, LoaderInterface $loader, array $config)
     {
         if (false === $config['mailer']['enabled']) {
-            return;
-        }
-
-        try {
-            $mailerDefinition = $container->findDefinition('mailer');
-        } catch (ServiceNotFoundException $e) {
-            throw new \InvalidArgumentException('To enable mail reporting you have to install the "swiftmailer/swiftmailer" or "symfony/mailer".');
+            $config['mailer'] = [
```

Comment:
What is the reason for keeping this?

Reply:
The compiler pass needs to know if the mailer configuration is enabled in order to load the services.

It looks cleaner to add a parameter with the mailer.enabled configuration rather than use `$container->getExtensionConfig('liip_monitor')` because the compiler pass does not receive the config array.

The rest of the mailer configuration should not be added as parameters if it is not enabled, so I replace the entire array to have just the enabled key and draw upon the "for" from below.

What's the Symfony best practice to accomplish this?

Thanks

Reply:
Ok, I see. What about just returning early if it isn't enabled (like before) and not setting any params. The compiler pass checks if the parameter exists so this should still work as expected.